src/tests/class_tests/openms/source/MRMDecoy_test.cpp

```diff
@@ -93,6 +93,25 @@ START_SECTION((std::vector<std::pair<std::string::size_type, std::string> > find
 
 END_SECTION
 
+START_SECTION((std::vector<std::pair<std::string::size_type, std::string> > find_all_tryptic_and_term(std::string sequence)))
+{
+  MRMDecoy gen;
+
+  String sequence = "TRESTPEPTIKDE";
+  std::vector<pair<std::string::size_type, std::string> > tryptic_results = gen.find_all_tryptic_and_term(sequence);
+  std::vector<pair<std::string::size_type, std::string> > tryptic_control = boost::assign::list_of(std::make_pair(0, "T")) (std::make_pair(1, "R")) (std::make_pair(5, "P")) (std::make_pair(7, "P")) (std::make_pair(10, "K")) (std::make_pair(12, "E"));
```

Comment:
I am not sure this is really doing what it is supposed to, Trypsin cuts at `[K|R]` except when followed by `P` so this should really be

```
(std::make_pair(0, "T")
(std::make_pair(1, "R")
(std::make_pair(10, "K")
(std::make_pair(12, "E")
```

or am I missing something here?

Reply:
Due to their strong fragment ion signal, it may make sense to keep Prolines where they are, is this what is intended here?

Reply:
We adapted this function from the ``SpectraST`` decoy generator: http://pubs.acs.org/doi/full/10.1021/pr900947u

Reply:
gen.find_all_tryptic_and_term(sequence); should be finAllTrypticAndTerm
by the way, does this use the EnzymaticDigestion capabilities in the core library? And if not, are there specific reason?

Reply:
I can rename the function. Here, we don't do tryptic digestion, but only need to recover the indices of all K, R & P residues. We wrote the original function to represent the SpectraST implementation. I'm not sure whether it is worthwhile to port this to the full EnymaticDigestion class.

Reply:
Ok it just seemed similar to what tokenize_ etc. does and, thus, might help to reduce some code duplication (without having looked deeper into the code)

Reply:
ok, but lets call it something different than tryptic termini as it clearly also finds other AA which are not tryptic termini (such as P)

Reply:
Good suggestion. The functions are now called findFixedResidues and findFixedAndTermResidues.